File: backend/app/models/like.py
from app.extensions import db
from app.models import BaseModel

class Like(BaseModel):
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), primary_key=True)
    photo_id = db.Column(db.Integer, db.ForeignKey('photo.id'), primary_key=True)
    user = db.relationship("User", back_populates="likes")
    photo = db.relationship("Photo", back_populates="likes")

File: app/models/like.rb
# == Schema Information
#
# Table name: likes
#
#  id         :integer          not null, primary key
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  photo_id   :integer          not null
#  user_id    :integer          not null
#
# Indexes
#
#  index_likes_on_photo_id  (photo_id)
#  index_likes_on_user_id   (user_id)
#
# Foreign Keys
#
#  photo_id  (photo_id => photos.id)
#  user_id   (user_id => users.id)
#
class Like < ApplicationRecord
  belongs_to :user
  belongs_to :photo, counter_cache: true

  validates :user_id, uniqueness: { scope: :photo_id, message: "has already liked this photo" }
end
